docs/recommendations.md

```diff
@@ -0,0 +1,88 @@
+Invocation flow for a recommendations function annotated in [`Common.RecommendationsFunction`](../vocabularies/Common.md#RecommendationsFunction)
+
+```mermaid
+sequenceDiagram
+  actor client as Fiori client
+  box ABAP
+    participant server as OData service
+    participant task as New task
+  end
+  participant AI as BTP recommendations service
+  activate server
+  note over server: sticky session
+  client ->> server: POST $batch
+  server ->> server: PATCH SalesOrder('A')
+  par
+    server ->> server: GET SalesOrder('A')?$expand=... for side effects
+  and
+    server ->> server: GET SalesOrder('A')/ns.Recommendations?Targets=...<br>Prefer: respond-async
+    server ->>+ task: call function ... starting new task
+  end
+  server -->> client: $batch response with<br>Location header for recommendations
+  task ->>+ AI: compute recommendations
+  client ->>+ server: GET «Location header for recommendations»
+  note right of server: outside sticky session
+  par
+    loop until successful lookup
```

Comment:
> Will applications have a choice whether to use "long polling" or "repeated poll requests"? If so, it would be good for the client to know which is used (at least if a sticky session is used as well).

This document explains both methods. Which ones we support and how we inform the client is not part of this PR.

Reply:
> Will applications have a choice whether to use "long polling" or "repeated poll requests"?

The server could use long polling if called outside a sticky session (that is, without `sap-contextid` header) and short polling otherwise. Then it's up to the client whether to poll inside or outside the session.

Reply:
But these SAP specific additions will have to be defined somewhere. So essentially, we will need to see which way we will go in the productive implementation. Then there may be the need for some additional specification.